gnosis/src/main/scala/edu/uci/ics/cloudberry/gnosis/IGeoIndex.scala

```diff
@@ -41,4 +41,32 @@ class USGeoJSONIndex() extends IGeoIndex {
   override def search(envelope: Envelope): Seq[IUSGeoJSONEntity] = {
     index.query(envelope).asScala.map(item => entities(item.asInstanceOf[Int]))
   }
+}
+
+class NYGeoJSONIndex() extends IGeoIndex {
```

Comment:
Failed to find a way out. I tried to move loadShape to IGeoIndex and change IUSGeoJSONEntity into IEntity. Declare entities in IGeoIndex as ArrayBuffer[IEntity].
It can't compile successfully. @JavierJia Help!!!